modules/local/ensembl_geneset_download.nf

```diff
@@ -0,0 +1,68 @@
+// Module that downloads all necessary geneset files from Ensembl.
+// The module checks that the MD5 checksums match before releasing the data.
+// It also uncompresses the files, since we want bgzip compression.
+process ENSEMBL_GENESET_DOWNLOAD {
+    tag "${assembly_accession}|${geneset_version}"
+    label 'process_single'
+
+    conda (params.enable_conda ? "bioconda::wget=1.18" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gnu-wget:1.18--h7132678_6' :
+        'quay.io/biocontainers/gnu-wget:1.18--h7132678_6' }"
+
+    input:
+    tuple val(analysis_dir), val(ensembl_species_name), val(assembly_accession), val(geneset_version)
+
+    output:
+    tuple val(meta), env(ANNOTATION_METHOD), path("*-cdna.fa")    , emit: cdna
+    tuple val(meta), env(ANNOTATION_METHOD), path("*-cds.fa")     , emit: cds
+    tuple val(meta), env(ANNOTATION_METHOD), path("*-genes.gff3") , emit: gff
+    tuple val(meta), env(ANNOTATION_METHOD), path("*-pep.fa")     , emit: pep
+    path  "versions.yml"                                          , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+
+    // e.g. https://ftp.ensembl.org/pub/rapid-release/species/Agriopis_aurantiaria/GCA_914767915.1/geneset/2021_12/Agriopis_aurantiaria-GCA_914767915.1-2021_12-cdna.fa.gz
+    def ftp_path = params.ftp_root + "/" + ensembl_species_name + "/" + assembly_accession + "/geneset/" + geneset_version
+    def remote_filename_stem = ensembl_species_name + "-" + assembly_accession + "-" + geneset_version
+
+    // id will be added later
+    meta = [
+        outdir : analysis_dir,
+        assembly_accession : assembly_accession,
+        geneset_version: geneset_version,
+    ]
```

Comment:
Try to implement this in the `download` subworkflows. Suggested input:
`tuple val(meta), val(ftp_path), val(remote_filename_stem)`

Reply:
I think you're right, and it especially makes sense because I heavily edit `meta` in the download subworkflow anyway !